src/transformers/models/funnel/modeling_tf_funnel.py

```diff
@@ -485,10 +486,15 @@ def relative_positional_attention(self, position_embeds, q_head, context_len, cl
                 "bind,jd->bnij", q_r_attention_2, omega
             )
         else:
-            shift = 2 if q_head.shape[1] != context_len else 1
             # Notations from the paper, appending A.2.1, final formula (https://arxiv.org/abs/2006.03236)
             # Grab the proper positional encoding, shape max_rel_len x d_model
-            r = position_embeds[self.block_index][shift - 1]
+            # shift = 2 if shape_list(q_head)[1] != context_len else 1
+            if shape_list(q_head)[1] != context_len:
+                shift = 2
+                r = position_embeds[self.block_index][1]
+            else:
+                shift = 1
+                r = position_embeds[self.block_index][0]
```

Comment:
Looks like the shift variable isn't used after? In this case, we can just remove it.

Reply:
It is used after see line 508.